lsp/src/main/java/org/aya/lsp/utils/ForIDEA.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2020-2022 Yinsen (Tesla) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.lsp.utils;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.Serializable;
+
+/**
+ * Special gift for Intellij IDEA plugin. Usually used in response models to skip json serialization of certain data.
+ * @see org.aya.lsp.models.HighlightResult.Symbol
+ */
+public final class ForIDEA<T> implements Serializable {
```

Comment:
Isn't this `Ref<T>`?

Reply:
The inner instance of `Ref<T>` is not transient. And the purpose of `ForIDEA` is skipping serialization when serving VSCode

Reply:
I would suggest asking kala-common to implement a transient Ref, what do you think

Reply:
I agree with you. /cc @Glavo